packages/terra-application-layout/src/header/_ApplicationHeader.jsx

```diff
@@ -0,0 +1,253 @@
+  /**
+   * Internationalization object with translation APIs. Provided by `injectIntl`.
+   */
+  intl: intlShape,
+};
+
+class ApplicationHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleUtilityDiscloseRequest = this.handleUtilityDiscloseRequest.bind(this);
+    this.handleUtilityPopupCloseRequest = this.handleUtilityPopupCloseRequest.bind(this);
+    this.handleUtilityOnChange = this.handleUtilityOnChange.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.renderToggle = this.renderToggle.bind(this);
+    this.renderAppName = this.renderAppName.bind(this);
+    this.renderExtensions = this.renderExtensions.bind(this);
+    this.renderNavigation = this.renderNavigation.bind(this);
+    this.renderUtilities = this.renderUtilities.bind(this);
+    this.renderUtilitiesPopup = this.renderUtilitiesPopup.bind(this);
+
+    this.state = { utilityComponent: undefined };
+  }
+
+  setContentNode(node) {
+    this.contentNode = node;
+  }
+
+  getTargetRef() {
+    if (this.contentNode) {
+      return this.contentNode.querySelector('[data-application-header-utility]');
+    }
+    return undefined;
+  }
+
+  handleUtilityDiscloseRequest(utility) {
+    this.setState({
+      utilityComponent: React.cloneElement(utility, { onRequestClose: this.handleUtilityPopupCloseRequest }),
+    });
+  }
+
+  handleUtilityPopupCloseRequest() {
+    if (this.state.utilityComponent) {
+      this.setState({ utilityComponent: undefined });
+    }
+  }
+
+  handleUtilityOnChange(event, key) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, key, app && app.disclose);
+  }
+
+  renderToggle() {
+    const { layoutConfig, intl } = this.props;
+
+    if (layoutConfig.toggleMenu) {
+      return (
+        <div className={cx('toolbar-toggle')}>
+          <button
+            className={cx('toggle-button')}
+            aria-label={intl.formatMessage({ id: 'Terra.applicationLayout.applicationHeader.menuToggleLabel' })}
+            onClick={layoutConfig.toggleMenu}
+          >
+            <IconMenu />
+          </button>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderAppName() {
+    const { nameConfig } = this.props;
+
+    if (nameConfig.accessory || nameConfig.title) {
+      return (
+        <ApplicationHeaderName accessory={nameConfig.accessory} title={nameConfig.title} />
+      );
+    }
+
+    return null;
+  }
+
+  renderNavigation(isCompact) {
+    const { applicationLinks } = this.props;
+
+    if (!isCompact) {
+      if (applicationLinks && applicationLinks.length) {
+        return (
+          <ApplicationTabs links={applicationLinks} />
+        );
+      }
+
+      return null;
+    }
+
+    /**
+     * When compact, the navigation region of the header renders the application name component instead.
+     */
+    return this.renderAppName();
```

Comment:
Is there a reason we toggle where `this.renderAppName();` is rendered, either in logo or navigation region?

Reply:
I'll enhance the comment.

At compact sizes, the logo region within the ApplicationHeaderLayout is too small to use, so we instead render within the navigation region which renders with a larger width.